src/v/iceberg/tests/update_schema_action_test.cc

```diff
@@ -183,21 +183,25 @@ TEST_F(UpdateSchemaActionTest, TestNewMultipleSchemas) {
       0);
 }
 
-TEST_F(UpdateSchemaActionTest, TestInvalidSchema) {
-    // Removing columns is not yet supported.
+TEST_F(UpdateSchemaActionTest, TestErroredTransaction) {
+    auto tx = transaction::make_with_error(
+      create_table(), action::errc::unexpected_state);
+
+    // Adding a new schema on an already errored transaction will fail
+    auto res = tx.set_schema(make_schema(12345)).get();
+    ASSERT_TRUE(res.has_error());
+    ASSERT_TRUE(tx.error().has_value());
+}
+
+TEST_F(UpdateSchemaActionTest, TestRemoveColumn) {
+    // Removing columns is allowed
     transaction tx(create_table());
     auto new_schema = make_schema(12345);
     new_schema.schema_struct.fields.pop_back();
 
     auto res = tx.set_schema(std::move(new_schema)).get();
-    ASSERT_TRUE(res.has_error());
-    ASSERT_TRUE(tx.error().has_value());
-
-    // Adding a new schema after attempting a bad update also fails.
-    auto new_new_schema = make_schema(12345, 1);
-    res = tx.set_schema(std::move(new_new_schema)).get();
-    ASSERT_TRUE(res.has_error());
-    ASSERT_TRUE(tx.error().has_value());
+    ASSERT_FALSE(res.has_error());
+    ASSERT_FALSE(tx.error().has_value());
```

Comment:
Can we assert that the last column id is what we expect?

Reply:
ya good catch

Reply:
done